python/cuspatial/cuspatial/core/geodataframe.py

```diff
@@ -120,6 +126,35 @@ def _copy_type_metadata(
 
         return type_copied
 
+    def _slice(self: T, arg: slice) -> T:
+        """
+        Overload the _slice functionality from cudf's frame members.
+        """
+        # Collect the Geometry columns and slice them separately
+        columns_mask = cudf.Series(self.columns)
```

Comment:
cudf does not guarantee performance with wide table. Neither should geodataframe. There's no need to use device to handle the column name list.